runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java

```diff
@@ -56,6 +56,13 @@ public void incrementLapCount(Client client)
 		int nextLevel = client.getRealSkillLevel(Skill.AGILITY) + 1;
 		int remainingXp = nextLevel <= Experience.MAX_VIRT_LEVEL ? Experience.getXpForLevel(nextLevel) - currentExp : 0;
 
+		//If next level is reached, count to the level after
+		if (remainingXp < 0)
+		{
+			nextLevel += 1;
+			remainingXp = nextLevel <= Experience.MAX_VIRT_LEVEL ? Experience.getXpForLevel(nextLevel) - currentExp : 0;
```

Comment:
Make a function for this? It's exactly the same code as L57.

Reply:
Changed it to while loop, as it occurred to me that in things like DMM one might level up more than one level upon course completion